design/one-pager-moving-packages.md

```diff
@@ -0,0 +1,462 @@
+2. Replicating a dependency to a different, potentially private, registry
+
+Handling private registry authentication for dependencies isn't in scope for
+this design. It's covered by the [package `ImageConfig` proposal][imageconfig].
+
+## Proposal
+
+I propose we do two things:
+
+1. Allow packages to specify that they replace another package
+2. Allow packages to specify partially qualified dependencies
+
+These proposed changes correspond to the goals above. The first addresses
+moving dependencies, while the second addresses replicating dependencies.
+
+### Moving a Package
+
+I propose we support moving a package by adding a `replaces` stanza to package
+metadata.
+
+For example, assume the maintainers of `provider-family-aws` move it from
+`xpkg.upbound.io/upbound/provider-family-aws` to
+`xpkg.upbound.io/crossplane-contrib/provider-family-aws`. The maintainers would
+add the following `replaces` stanza to the moved provider:
+
+```yaml
+apiVersion: meta.pkg.crossplane.io/v1
+kind: Provider
+metadata:
+  name: provider-family-aws
+spec:
+  replaces:
+  - xpkg.upbound.io/upbound/provider-family-aws
+```
+
+This tells Crossplane that the `crossplane-contrib/provider-family-aws` package
+replaces the `upbound/provider-family-aws` package.
+
+When the revision controller encounters a package with a replaces stanza, it'll
+deactivate the replaced package(s) before installing the new package.
+
+When the replaced package is deactivated, the new package can be installed
+without a conflict. A deactivated package [owns][owner-ref] but doesn't
+[control][controller-ref] its payload (e.g. a Provider's MR CRDs). This
+eliminates the controller reference conflict.
+
+A deactivated package exists, but in an inactive state. Specifically:
+
+* It appears under (e.g.) `kubectl get providers`
+* At least one revision appears under (e.g.) `kubectl get providerrevisions`
+* The package's `spec.revisionActivationPolicy` is set to `Manual`
+* All revisions have `spec.desiredState` set to `Inactive`
+* It has no runtime - e.g. no Deployment of a provider's runtime binary
+* It owns but doesn't control its package payload - e.g. a provider's MR CRDs
+
+Crucially, deactivating a package doesn't remove its payload. For example
+deactivating a provider doesn't remove its MR CRDs like uninstalling it would.
+The new package shares ownership (in the [owner reference][owner-ref] sense)
+with the replaced package, until the replaced package is manually uninstalled.
+This allows the package manager to seamlessly handover ownership and control of
+a package's payload.
+
+To deactivate a replaced packaged, the revision controller will:
+
+1. Map the replaced source to its active package revision using the Lock
+2. Map the package revision to its package using its owner reference
+3. Set the package's `spec.revisionActivationPolicy` to `Manual`
+4. Set the package revision's `spec.desiredState` to `Inactive`
```

Comment:
While testing the dependency version upgrade change, I noticed that, when a package revision becomes inactive, package revision reconciler removes it from the lock object. Which means in a small period of time (until the new package revision is created) lock reconciler assumes that the dependency is not installed and tries to reinstall it. 

For the PR I'm currently working on, I plan to list all the packages and get the kind, source and version information to find out if a package is installed or not. Because, relying on only the lock object leads errors.

Reply:
Got it, that's good to know thank you!

I'll update the doc and do the same.